src/components/TodoFooter/TodoFooter.tsx

```diff
@@ -0,0 +1,88 @@
+import classNames from 'classnames';
+import { useContext, useMemo } from 'react';
+import { Status } from '../../types/Status';
+import { TodoContext } from '../TodoContext';
+import { deleteTodo } from '../../api/todos';
+import { ErrorContext } from '../ErrorContext';
+import { GlobalLoader } from '../../types/GlobalLoader';
+
+type Props = {
+  status: Status;
+  onStatusChange: (status: Status) => void;
+  onGlobalLoaderChange: (globalLoader: GlobalLoader) => void;
+};
+
+export const TodoFooter: React.FC<Props> = (props) => {
+  const {
+    status,
+    onStatusChange,
+    onGlobalLoaderChange,
+  } = props;
+
+  const { todos, setTodos } = useContext(TodoContext);
+  const { setError } = useContext(ErrorContext);
+
+  const uncomplitedTodos = useMemo(() => todos
+    .filter(({ completed }) => !completed), [todos]);
+
+  const handleComplDelete = () => {
+    onGlobalLoaderChange(GlobalLoader.Completed);
+
+    const deletedTodos: Promise<number>[] = [];
+
+    todos.forEach(({ id, completed }) => {
+      if (completed) {
+        deletedTodos.push(deleteTodo(id)
+          .then(() => id)
+          .catch(error => {
+            throw error;
+          }));
+      }
+    });
+
+    Promise.all(deletedTodos)
+      .then((res) => {
+        setTodos(prevState => prevState
+          .filter(todo => !res.includes(todo.id)));
+      })
+      .catch(() => setError('Unable to delete a todo'))
+      .finally(() => onGlobalLoaderChange(GlobalLoader.Non));
+  };
+
+  return (
+    <footer className="todoapp__footer">
+      <span className="todo-count">
+        {`${uncomplitedTodos.length} item${uncomplitedTodos.length === 1 ? '' : 's'} left`}
```

Comment:
To improve readability, it is worth transferring the definition of the text to a separate variable.